src/cg.rs

```diff
@@ -35,22 +46,55 @@ impl CGImpl {
             view.addSubview_(subview); // retains subview (+1) = 2
             let _: () = msg_send![subview, release]; // releases subview (-1) = 1
         }
-        Ok(Self { layer, window })
+        let color_space = CGColorSpace::create_device_rgb();
+        Ok(Self {
+            layer,
+            window,
+            color_space,
+            width: 0,
+            height: 0,
+        })
     }
 
-    pub(crate) unsafe fn set_buffer(&mut self, buffer: &[u32], width: u16, height: u16) {
-        let color_space = CGColorSpace::create_device_rgb();
-        let data =
-            unsafe { std::slice::from_raw_parts(buffer.as_ptr() as *const u8, buffer.len() * 4) }
-                .to_vec();
-        let data_provider = CGDataProvider::from_buffer(Arc::new(data));
+    pub fn resize(&mut self, width: u32, height: u32) -> Result<(), SoftBufferError> {
```

Comment:
I'd suggest to use `NonZeroU16` and that way we could get rid of new error variant all together, though, the `Result` will still stay here. Or `NonZeroU32`.

Reply:
There's an issue open about the fact the existing API only allows 16-bit values: https://github.com/rust-windowing/softbuffer/issues/13

I don't really expect anyone needs windows that large currently or in the foreseeable future, but it's probably best to stick with 32-bit if that's what modern backends support, and what winit, glutin, etc. use.

Reply:
Though it could be a panic instead of an error.

Reply:
You can't even fit into memory such windows, they'll take 128GB for their frame buffer...